### PR TITLE
Added functionality to upload file to warroom

### PR DESCRIPTION
**Pull request by ronaldeddings**
This commit enables demisto-py to upload files to warroom and also an exmaple of how to use the new feature

### CODE REVIEW

File: demisto/client.py [hunk 1]
```diff
@@ -33,7 +34,7 @@ def __init__(self, apiKey, server, username=None, password=None):
         if not apiKey:
             self.xsrf = True
             try:
-                r = self.session.get(server, verify=False)
+                response = self.session.get(server, verify=False)
```
Comment (slavikm) [inline demisto/client.py]: Need to change below as well

File: demisto/client.py [hunk 2]
```diff
@@ -43,22 +44,24 @@ def __init__(self, apiKey, server, username=None, password=None):
...
             self.apiKey = apiKey
 
-    def req(self, method, path, data):
-        h = {"Accept": "application/json",
-             "Content-type": "application/json"}
+    def req(self, method, path, data,files=None):
+        if files:
+            headers = {"Accept": "application/json"}
+        else:
+            headers = {"Accept": "application/json","Content-type": "application/json"}
```
Comment (slavikm) [inline demisto/client.py]: Space between commas

File: demisto/client.py [hunk 3]
```diff
@@ -43,22 +44,24 @@ def __init__(self, apiKey, server, username=None, password=None):
             self.xsrf = False
             self.apiKey = apiKey
 
-    def req(self, method, path, data):
-        h = {"Accept": "application/json",
-             "Content-type": "application/json"}
+    def req(self, method, path, data,files=None):
```
Comment (slavikm) [inline demisto/client.py]: Space between commas

File: demisto/client.py [hunk 4]
```diff
@@ -43,22 +44,24 @@ def __init__(self, apiKey, server, username=None, password=None):
...
         if self.xsrf:
-            h[DemistoClient.XSRF_TOKEN_KEY] = self.token
+            headers[DemistoClient.XSRF_TOKEN_KEY] = self.token
         else:
-            h[DemistoClient.AUTHORIZATION] = self.apiKey
+            headers[DemistoClient.AUTHORIZATION] = self.apiKey
         try:
             if self.session:
-                r = self.session.request(method, self.server+path, headers=h, verify=False, json=data)
+                response = self.session.request(method, self.server+path, headers=headers, verify=False, json=data,files=files)
```
Comment (slavikm) [inline demisto/client.py]: space

File: demisto/client.py [hunk 5]
```diff
@@ -83,7 +86,19 @@ def CreateIncident(self, inc_name, inc_type, inc_severity, inc_owner, inc_labels
 
     def SearchIncidents(self, page, size, query):
         data = {'filter': {'page': page, 'size': size, 'query': query, 'sort': [{'field':'id', 'asc': False}]}}
-        r = self.req("POST", "incidents/search", data)
-        if r.status_code != 200:
+        response = self.req("POST", "incidents/search", data)
+        if response.status_code != 200:
```
Comment (slavikm) [inline demisto/client.py]: If you're changing, might as well change to built-in constant instead of 200

File: demisto/client.py [hunk 6]
```diff
@@ -83,7 +86,19 @@ def CreateIncident(self, inc_name, inc_type, inc_severity, inc_owner, inc_labels
...
             raise RuntimeError('Error searching incidents - %d (%s)' % (r.status_code, r.reason))
-        return r.json()
+        return response.json()
+
+    def UploadFileToWarroom(self,inc_id,filePath):
+        files = None
+        with open(filePath,"rb") as fileObject:
```
Comment (slavikm) [inline demisto/client.py]: space

File: demisto/client.py [hunk 7]
```diff
@@ -83,7 +86,19 @@ def CreateIncident(self, inc_name, inc_type, inc_severity, inc_owner, inc_labels
...
         data = {'filter': {'page': page, 'size': size, 'query': query, 'sort': [{'field':'id', 'asc': False}]}}
-        r = self.req("POST", "incidents/search", data)
-        if r.status_code != 200:
+        response = self.req("POST", "incidents/search", data)
+        if response.status_code != 200:
             raise RuntimeError('Error searching incidents - %d (%s)' % (r.status_code, r.reason))
-        return r.json()
+        return response.json()
+
+    def UploadFileToWarroom(self,inc_id,filePath):
```
Comment (slavikm) [inline demisto/client.py]: space

File: demisto/client.py [hunk 8]
```diff
@@ -11,6 +11,7 @@
 from requests import Session
 from requests.packages.urllib3.exceptions import InsecureRequestWarning
 from requests.packages.urllib3 import disable_warnings
+import ntpath
```
Comment (slavikm) [inline demisto/client.py]: please use `os.path` and not `ntpath` as it is generic

File: upload_file_to_warroom_example.py [hunk 9]
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env python2.7
+
+# example
+# An example of helper utility to do stuff on multiple incidents based on filter
+#
+# Stuff can be closing incidents, running a command, changing type or changing playbook
+#
+# Author:       Slavik Markovich
```
Comment (slavikm) [inline upload_file_to_warroom_example.py]: Don't remember I wrote this

File: upload_file_to_warroom_example.py [hunk 10]
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env python2.7
+
+# example
+# An example of helper utility to do stuff on multiple incidents based on filter
```
Comment (slavikm) [inline upload_file_to_warroom_example.py]: update comment

File: upload_file_to_warroom_example.py [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+# Author:       Slavik Markovich
+# Version:      1.0
+#
+
+import json
+import argparse
+import demisto
+
+def options_handler():
+    parser = argparse.ArgumentParser(description='Utility for batch action on incidents')
```
Comment (slavikm) [inline upload_file_to_warroom_example.py]: update

File: demisto/client.py (hunk 1 above)
Comment (secdevopsai) [reply demisto/client.py]: done

File: demisto/client.py (hunk 2 above)
Comment (secdevopsai) [reply demisto/client.py]: done

File: demisto/client.py (hunk 3 above)
Comment (secdevopsai) [reply demisto/client.py]: done

File: demisto/client.py (hunk 4 above)
Comment (secdevopsai) [reply demisto/client.py]: done

File: demisto/client.py (hunk 5 above)
Comment (secdevopsai) [reply demisto/client.py]: done

File: demisto/client.py (hunk 7 above)
Comment (secdevopsai) [reply demisto/client.py]: done

File: demisto/client.py (hunk 8 above)
Comment (secdevopsai) [reply demisto/client.py]: done

File: upload_file_to_warroom_example.py (hunk 9 above)
Comment (secdevopsai) [reply upload_file_to_warroom_example.py]: done

File: upload_file_to_warroom_example.py (hunk 10 above)
Comment (secdevopsai) [reply upload_file_to_warroom_example.py]: done

File: upload_file_to_warroom_example.py (hunk 11 above)
Comment (secdevopsai) [reply upload_file_to_warroom_example.py]: done

File: demisto/client.py (hunk 6 above)
Comment (slavikm) [reply demisto/client.py]: Space

File: demisto/client.py (hunk 7 above)
Comment (slavikm) [reply demisto/client.py]: Don't see it...

**Comment by secdevopsai** [comment]
@slavikm my second commit has the changes you requested

**Comment by meirwah** [comment]
@slavikm can you review the changes?

**Comment by amshamah419** [comment]
We are releasing a V2 of the SDK. Closing out this PR.

**Pull request closed without merging**